java/ingest/ingest-runner/src/test/java/sleeper/ingest/impl/IngestCoordinatorUsingDirectWriteBackedByArrayListIT.java

```diff
@@ -86,121 +91,174 @@ public class IngestCoordinatorUsingDirectWriteBackedByArrayListIT {
     private final InstanceProperties instanceProperties = createTestInstanceProperties();
     private final String dataBucketName = instanceProperties.get(DATA_BUCKET);
     private final TableProperties tableProperties = createTestTablePropertiesWithNoSchema(instanceProperties);
+    private final Instant stateStoreUpdateTime = Instant.parse("2023-08-08T11:20:00Z");
+    private final RecordGenerator.RecordListAndSchema recordListAndSchema = RecordGenerator.genericKey1D(
+            new LongType(),
+            LongStream.range(-100, 100).boxed().collect(Collectors.toList()));
+    private final PartitionTree tree = new PartitionsBuilder(recordListAndSchema.sleeperSchema)
+            .rootFirst("root")
+            .splitToNewChildren("root", "left", "right", 0L)
+            .buildTree();
+    private final StateStore stateStore = createStateStore(recordListAndSchema.sleeperSchema);
 
     @BeforeEach
-    public void before() {
+    public void before() throws StateStoreException {
         s3.createBucket(instanceProperties.get(DATA_BUCKET));
         new S3StateStoreCreator(instanceProperties, dynamoDB).create();
+        stateStore.initialise(tree.getAllPartitions());
+        stateStore.fixTime(stateStoreUpdateTime);
     }
```

Comment:
I can't tell how these scenarios are different depending on the file writing strategy, since it's not asserting on the actual files, just the references.

How does the behaviour of overflowing memory or the local store change depending on the file writing strategy? Is it worth testing that, or should we maybe test against each of the PartitionFileWriter implementations in IngestCoordinatorFileWritingStrategyIT? Is it worth testing each file writing strategy against each PartitionFileWriter in the first place?